src/generator/models.ts

```diff
@@ -58,6 +58,7 @@ export async function generateModels(session: Session<CodeModel>): Promise<strin
   }
   if (needsJSONPopulate) {
     text += 'func populate(m map[string]interface{}, k string, v interface{}) {\n';
+    text += '\tif v == nil {\n\t\treturn\n\t}\n';
     text += '\tif azcore.IsNullValue(v) {\n';
```

Comment:
Would like to understand why both of these checks are necessary?

Reply:
The `v == nil` check is for types that have an `interface{}` field type(s).  In that case, if the field is `nil` then `v` will also be `nil` which causes `azcore.IsNullValue()` to panic.